llvm/lib/Target/SPIRV/SPIRVOpenCLBIFs.cpp

```diff
@@ -172,6 +180,14 @@ static StringMap<GroupOperation::GroupOperation> GroupOperationMap({
 static bool genBarrier(MachineIRBuilder &MIRBuilder, unsigned opcode,
     const SmallVectorImpl<Register> &OrigArgs, SPIRVTypeRegistry *TR);
 
+bool llvm::getSpirvBuilInIdByName(StringRef Name, BuiltIn::BuiltIn &BI) {
```

Comment:
Not blocking this PR, just a note.

I think such an approach to the interface design, I mean return byref param + bool return value is far from good and I remember I saw it in multiple places across the backend (e.g. in the DT I originally designed too) and I suggest we should replace that. I was thinking about returning a pointer which is either good or nullptr. Or std::optional would help as well, but AFAIK llvm trunk is still on C++14 so not an option for now

Reply:
Thanks, agree in general case. BuiltIn::BuiltIn is a numerical value so I didn't want to allocate memory and return a pointer to it. The natural way for this case would be to introduce and return "BuiltInLast" value for the enum if the string is not recognized as a builtin variable name.